### DISCUSSION TITLE
x-init executing my function?

### DISCUSSION THREAD

**Discussion by VictorioBerra** [alpinejs/alpine#4206] [1. Help]
`x-data="global" x-init="myVar = function(){ console.log('hello world') }"`

Console log runs.

I need to set global.myVar from my children so global can invoke them as needed and the children can provide the functionality.

**Comment by VictorioBerra** [comment]
I refactored and ended up just putting a child div in and assigning it one level deeper using x-data.

**Comment by ekwoka** [comment]
the expression is evaluated if it returns a function, which this does (assignments evaluate to whatever the assignment is).

This is probably not a good way to do whatever it is trying to do though.
